include/spdlog/sinks/android_sink.h

```diff
@@ -0,0 +1,101 @@
+/* spdlog - an extremely fast and easy to use c++11 logging library.     */
+/* Copyright (c) 2014 Gabi Melman.                                       */
+/* Copyright (c) 2015 Ruslan Baratov.                                    */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#pragma once
+
+#if defined(__ANDROID__)
+
+#include <mutex>
+#include "base_sink.h"
+#include "../details/null_mutex.h"
+
+#include <android/log.h>
+
+namespace spdlog
+{
+namespace sinks
+{
+/*
+* Android sink (logging using __android_log_write)
+*/
+template<class Mutex>
+class base_android_sink : public base_sink < Mutex >
+{
+public:
+    explicit base_android_sink(std::string tag="spdlog"): _tag(tag)
+    {
+    }
+
+    void flush() override
+    {
+    }
+
+protected:
+    void _sink_it(const details::log_msg& msg) override
+    {
+        const android_LogPriority priority = convert_to_android(msg.level);
+        const int expected_size = msg.formatted.size();
+        const int size = __android_log_write(
+            priority, _tag.c_str(), msg.formatted.c_str()
+        );
+        if (size > expected_size)
+        {
+            // Will write a little bit more than original message
+        }
+        else
+        {
+            throw spdlog_ex("Send to Android logcat failed");
+        }
+    }
+
+private:
+    static android_LogPriority convert_to_android(spdlog::level::level_enum level)
+    {
+        switch(level)
```

Comment:
> simple lookup instead of recalculating over and over again

`simple lookup` mean loading value from memory and `recalculating over and over again` mean quite trivial arithmetic on registers. Just imagine if `spdlog::level::FOO` equals to `ANDROID_LOG_FOO`, i.e. this function just do nothing and return `level` argument. Anyway I think both operations take very small time so I doubt we can take any adequate benchmarks here.

> Also maybe if the test used emerg() instead of info() youd see a bigger difference.

nope, still the same

Also you may find interesting this test: https://gist.github.com/ruslo/f0bb18c73f3d08ea544b
As you can see in case of `boo` (with switch) compiler is smart enough to figure out that result is 0x76 = 0x34 + 0x42, but not in case of `foo` (with loading from memory).